pkg/processor/worker/worker.go

```diff
@@ -44,8 +52,12 @@ type Worker struct {
 	structuredCloudEvent cloudevent.Structured
 	binaryCloudEvent     cloudevent.Binary
 	eventTime            *time.Time
-	isDrained            atomic.Bool
-	drainedLock          sync.Mutex
+	state                StateWithLock
+}
+
+type StateWithLock struct {
+	State
+	sync.Mutex
 }
```

Comment:
move up below `State` consts

Reply:
@TomerShor this is impossible, because we cannot change constants in go.

Reply:
It is very possible to just move the block :) 

```golang
type State string

const (
	StateAvailable  State = "available"
	StateDrained    State = "drain"
	StateTerminated State = "terminated"
)

type StateWithLock struct {
	State
	sync.Mutex
}
```

Reply:
@TomerShor misunderstood, will move